go/adbc/driver/flightsql/record_reader.go

```diff
@@ -32,38 +32,48 @@ import (
 )
 
 type reader struct {
-	refCount int64
-	schema   *arrow.Schema
-	ch       chan arrow.Record
-	rec      arrow.Record
-	err      error
+	refCount   int64
+	schema     *arrow.Schema
+	chs        []chan arrow.Record
+	curChIndex int
+	rec        arrow.Record
+	err        error
 
 	cancelFn context.CancelFunc
 }
 
 // kicks off a goroutine for each endpoint and returns a reader which
 // gathers all of the records as they come in.
 func newRecordReader(ctx context.Context, alloc memory.Allocator, cl *flightsql.Client, info *flight.FlightInfo, clCache gcache.Cache) (rdr array.RecordReader, err error) {
-	ch := make(chan arrow.Record, 5)
-	group, ctx := errgroup.WithContext(ctx)
-	ctx, cancelFn := context.WithCancel(ctx)
-
-	defer func() {
+	endpoints := info.Endpoint
+	numEndpoints := len(endpoints)
```

Comment:
put this closer to where it is used?

Reply:
It does need to be before we potentially mutate endpoints below but I'll move it down a bit